Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -153,35 +171,36 @@ public final class ViewAnnotationManager {
 
         for position in positions {
             validateAnnotation(byAnnotationId: position.identifier)
-            guard let annotationView = annotationViewsById[position.identifier] else {
+            guard let view = viewsById[position.identifier] else {
                 continue
             }
-            annotationView.translatesAutoresizingMaskIntoConstraints = true
-            annotationView.frame = CGRect(
+            view.translatesAutoresizingMaskIntoConstraints = true
+            view.frame = CGRect(
                 origin: position.leftTopCoordinate.point,
                 size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
             )
-            annotationView.setInternalVisibility(isHidden: false)
+            view.isHidden = false
+            expectedHiddenForId[position.identifier] = false
             visibleAnnotationIds.insert(position.identifier)
         }
 
-        let annotationsToHide = Set<String>(annotationViewsById.keys).subtracting(visibleAnnotationIds)
+        let annotationsToHide = Set<String>(viewsById.keys).subtracting(visibleAnnotationIds)
         for id in annotationsToHide {
             validateAnnotation(byAnnotationId: id)
-            annotationViewsById[id]?.setInternalVisibility(isHidden: true)
+            viewsById[id]?.isHidden = true
+            expectedHiddenForId[id] = true
         }
     }
 
     internal func validateAnnotation(byAnnotationId id: String) {
-        guard let annotationView = annotationViewsById[id] else { return }
-        // If the user explicitly removed the ViewAnnotation or it's wrapped view
-        // we need to remove it from our layout calculation
-        if annotationView.subviews.isEmpty || annotationView.superview == nil {
-            try? remove(annotationView)
+        guard let view = viewsById[id] else { return }
+        // If the user removed the view from it's superview, then we can remove it from our layout calculation
+        if view.superview == nil {
+            try? remove(view)
         }
-        // View is still considered for layout calculation, users should not modify the visibility of the wrapped view
-        if let wrappedView = annotationView.subviews.first, wrappedView.isHidden {
-            Log.warning(forMessage: "Visibility changed for wrapped view", category: "Annotations")
+        // View is still considered for layout calculation, users should not modify the visibility of view directly
+        if !suppressVisibilityWarnings && view.isHidden != expectedHiddenForId[id] {
```

Comment:
Do you want to check the view's alpha here too?

Reply:
Not sure. I think if we set the alpha to zero we still expect it to be part of the layout. Also it was decided on Android to ignore the zero alpha case, so probably it's easier if we don't handle it either.